<launch>
    <!-- human detection config -->
    <arg name="HUMAN_CAM_ID"/>
    <arg name="HUMAN_CONF_THRESHOLD" default="0.71"/>
    <arg name="HUMAN_AREA_THRESHOLD" default="0.125"/>
    <arg name="HUMAN_ENTER_MIN" default="1.0"/>
    <arg name="HUMAN_EXIT_MAX" default="0.5"/>
    <arg name="HUMAN_BOUNDS" default="0.0,0.0,1.0,1.0"/>

    <!-- bottle detection config -->
    <arg name="BOTTLE_CAM_ID"/>
    <arg name="BOTTLE_CONF_THRESHOLD" default="0.45"/>
    <arg name="BOTTLE_AREA_THRESHOLD" default="0.0"/>
    <arg name="BOTTLE_ENTER_MIN" default="0.25"/>
    <arg name="BOTTLE_EXIT_MAX" default="0.25"/>
    <arg name="BOTTLE_BOUNDS" default="0.25,0.0,0.4,1.0"/>

    <arg name="SHOW_CAM" default="false"/>
    
    <include file="$(find yolo_detector)/launch/yolo.launch">
        <arg name="NAME" value="human_detector"/>
        <arg name="CAM_ID" value="$(arg HUMAN_CAM_ID)"/>
        <arg name="CLASSIFIERS" value="person"/>
        <arg name="CONF_THRESHOLD" value="$(arg HUMAN_CONF_THRESHOLD)"/>
        <arg name="AREA_THRESHOLD" value="$(arg HUMAN_AREA_THRESHOLD)"/>
        <arg name="ENTER_MIN" value="$(arg HUMAN_ENTER_MIN)"/>
        <arg name="EXIT_MAX" value="$(arg HUMAN_EXIT_MAX)"/>
        <arg name="SHOW_CAM" value="$(arg SHOW_CAM)"/>
        <arg name="BOUNDARY" value="$(arg HUMAN_BOUNDS)"/>
    </include>

    <include file="$(find yolo_detector)/launch/yolo.launch">
        <arg name="NAME" value="bottle_detector"/>
        <arg name="CAM_ID" value="$(arg BOTTLE_CAM_ID)"/>
        <arg name="CLASSIFIERS" value="bottle"/>
        <arg name="CONF_THRESHOLD" value="$(arg BOTTLE_CONF_THRESHOLD)"/>
        <arg name="AREA_THRESHOLD" value="$(arg BOTTLE_AREA_THRESHOLD)"/>
        <arg name="ENTER_MIN" value="$(arg BOTTLE_ENTER_MIN)"/>
        <arg name="EXIT_MAX" value="$(arg BOTTLE_EXIT_MAX)"/>
        <arg name="SHOW_CAM" value="$(arg SHOW_CAM)"/>
        <arg name="BOUNDARY" value="$(arg BOTTLE_BOUNDS)"/>
    </include>

    <node name="detector_merge"
        pkg="handover_demo"
        type="detector_merge.py"
        output="screen"
        required="true"/>
</launch>